### PR TITLE
Merge development with master

### PR DESCRIPTION
**Pull request by Nicklason**
- Added message command

### CODE REVIEW

**Comment by idinium96** [comment]
You forgot to update `template.env` and `ecosystem.template.json` files.

**Comment by Gobot1234** [comment]
what happened here lol

File: src/classes/Commands.ts [hunk 1]
```diff
@@ -321,6 +325,79 @@ export = class Commands {
...
+                return;
+            } else if (!this.bot.friends.isFriend(recipient)) {
+                this.bot.sendMessage(steamID, 'I am not friends with the user.');
+                return;
+            }
+
+            const reply = parts[2];
+
+            // Send message to recipient
+            this.bot.sendMessage(recipient, 'Message from ' + (details ? details.player_name : 'admin') + ': ' + reply);
```
Comment (idinium96) [inline src/classes/Commands.ts:368]: This shouldn't be `(isAdmin ? isAdmin.player_name : 'admin')` yes?

Comment (Nicklason) [reply src/classes/Commands.ts:368]: No. You getFriend can return null so "admin" is just a placeholder if the name was not found.

Comment (idinium96) [reply src/classes/Commands.ts:368]: ah I see.. alright kinda confused a bit 👌🏻

**Comment by idinium96** [comment]
Nope.
[notworking1](https://user-images.githubusercontent.com/47635037/78509078-020b2d80-77be-11ea-96e9-0df73ce95c40.PNG)
[notworking2](https://user-images.githubusercontent.com/47635037/78509080-03d4f100-77be-11ea-83ea-7c62ba0f6943.PNG)

File: src/classes/Commands.ts [hunk 2]
```diff
@@ -321,6 +325,79 @@ export = class Commands {
...
+
+            if (!new SteamID(recipient).isValid()) {
+                this.bot.sendMessage(steamID, '"' + recipient + '" is not a valid steamid.');
+                return;
+            } else if (!this.bot.friends.isFriend(recipient)) {
+                this.bot.sendMessage(steamID, 'I am not friends with the user.');
+                return;
+            }
+
+            const reply = parts[2];
```
Comment (ZeusJunior) [inline src/classes/Commands.ts:365]: This only sends the first word of the reply

Comment (idinium96) [reply src/classes/Commands.ts:365]: This works just fine
https://github.com/Nicklason/tf2-automatic/pull/414/commits/5be3028d6178225b8571dbaaf0469c669af532ec#diff-3a17c3b76a1ef21dd7e714d8c8f0f4fbR357

**Comment by idinium96** [comment]
Here's my report:
[3  admin-to-recipient-working](https://user-images.githubusercontent.com/47635037/78512098-5d481a80-77d4-11ea-823c-b3a00ccfdc94.PNG)
[4  recipient-to-admin-notworking](https://user-images.githubusercontent.com/47635037/78512100-5f11de00-77d4-11ea-8057-1403dfc7d984.PNG)

**Comment by idinium96** [comment]
I can confirm it reads the else part (not admin), but the bot still not able to send a message to the admin.
[5  read-but-not-sending-to-admin](https://user-images.githubusercontent.com/47635037/78512559-43a8d200-77d8-11ea-9b5b-b2f83bb1ff90.PNG)

File: src/classes/Bot.ts [hunk 3]
```diff
@@ -187,14 +191,25 @@ export = class Bot {
...
-        });
+        const message: string = args.length === 2 ? args[0] : args[1];
+        const exclude: string[] = (args.length === 2 ? [] : args[2]).map(steamid => steamid.toString());
+
+        this.admins
+            .filter(steamID => exclude.includes(steamID.toString()))
+            .forEach(steamID => {
+                this.sendMessage(steamID, message);
+            });
     }
```
Comment (idinium96) [inline src/classes/Bot.ts:213]: I think `messageAdmins` is corrupted.
[messageAdmins-not-working](https://user-images.githubusercontent.com/47635037/78546670-62d34e00-7830-11ea-9d1a-556dfdfcdfee.PNG)

**Comment by idinium96** [comment]
let me try to run it once.

**Comment by idinium96** [comment]
Yay! 🎉🎊
[messageCommand-working-fine](https://user-images.githubusercontent.com/47635037/78947812-0f475580-7af9-11ea-88f1-258119b2af18.PNG)

File: src/classes/Commands.ts [hunk 4]
```diff
@@ -321,6 +325,94 @@ export = class Commands {
...
+            }
+
+            const recipentDetails = this.bot.friends.getFriend(recipientSteamID);
+
+            const reply = message.substr(message.toLowerCase().indexOf(recipient) + 18);
+
+            // Send message to recipient
+            this.bot.sendMessage(
+                recipient,
+                'Message from ' + (adminDetails ? adminDetails.player_name : 'admin') + ': ' + reply
```
Comment (idinium96) [inline src/classes/Commands.ts:374]: Which is better? 'admin' or 'owner'? 🤔

**Comment by idinium96** [comment]
🤔
ok

File: src/classes/Commands.ts (hunk 4 above)
Comment (Nicklason) [reply src/classes/Commands.ts:374]: Admin is better because that's what it is called in the config

**Pull request merged**